jmclient/jmclient/wallet_utils.py

```diff
@@ -697,6 +699,12 @@ def wallet_generate_recover(method, walletspath,
     return True
 
 
+def wallet_change_passphrase(walletservice):
+    passphrase = cli_get_wallet_passphrase_check(changing_passphrase=True)
```

Comment:
The functions starting `cli_*` are there to be used as callbacks for the CLI scripts; the idea is that you can use them as default callbacks, and then Qt (or any other future client interface) can pass in a different callback to achieve the same goal.

See e.g. the call to `wallet_generate_recover_bip39` in `wallet_generate_recover` (although I'm not sure now why they are not optional keyword arguments with defaults, but .. details).

Anyway if it isn't obvious I'm saying make the wallet passphrase check function an argument, with default `cli_get_wallet_passphrase_check`, so that adding this in Qt will be straightforward.

Reply:
@AdamISZ Is this below what you had in mind?

```
def wallet_change_passphrase(walletservice,
                             enter_wallet_passphrase_callback=cli_get_wallet_passphrase_check):
    passphrase = enter_wallet_passphrase_callback(changing_passphrase=True)
    walletservice.change_wallet_passphrase(passphrase)
    return True
```